### PR TITLE
Ruchitha-javaAssignment 2

### PR DESCRIPTION
**Pull request by Kukkadapu-Ruchitha**
week 2 assignment-abstracttion,interface,sorting,pairedsum

### CODE REVIEW

File: week2/AbstractExample/AbstractDemo.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+public class AbstractDemo {
+    public static void main(String[] args) {
+        Student r1=new Student();
+        r1.setName("Anu");
+        r1.setAddress("NewDelhi");
+        System.out.println("Name is :"+r1.getName());
```
Comment (VBhaskarachary) [inline week2/AbstractExample/AbstractDemo.java]: Please use Abstract class as reference for all student, doctor and teacher

File: week2/AbstractExample/Person.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+public abstract class Person {
+      String name;
+     String address;
```
Comment (VBhaskarachary) [inline week2/AbstractExample/Person.java]: Abstract class contain at least one implementation method or construct

File: week2/InterfaceExample/MyInterface.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        System.out.println("Manufacturing companies");
+    }
+
+    public static void main(String[] args) {
+        interfaceDemo id = new interfaceDemo();
+        id.company();
+        Bike1 b1 = new Bike1();
+        b1.company();
+        b1.speedLimit();
```
Comment (VBhaskarachary) [inline week2/InterfaceExample/MyInterface.java]: Use interface as reference for all Bike implemention class, pls follow the requirement as given restrict the speed limit of a bike to 90 KMPH

File: week2/Sort/SortingAndSearching.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import java.util.*;
+public class SortingAndSearching {
+    static List<Number> numbers=new ArrayList<>();
+     static Set<Number> noDuplicate=new HashSet<>();
+    public  List<Number> removeDuplicates(List<Number> numbers){
+        List<Number> originalList=new ArrayList<>();
+        for(Number number:numbers){
```
Comment (VBhaskarachary) [inline week2/Sort/SortingAndSearching.java]: Use proper Utility collection class as per requirement

File: week2/AbstractExample/AbstractDemo.java (hunk 1 above)
Comment (Kukkadapu-Ruchitha) [reply week2/AbstractExample/AbstractDemo.java]: fixed

File: week2/AbstractExample/Person.java (hunk 2 above)
Comment (Kukkadapu-Ruchitha) [reply week2/AbstractExample/Person.java]: fixed

File: week2/InterfaceExample/MyInterface.java (hunk 3 above)
Comment (Kukkadapu-Ruchitha) [reply week2/InterfaceExample/MyInterface.java]: fixed

File: week2/Sort/SortingAndSearching.java (hunk 4 above)
Comment (Kukkadapu-Ruchitha) [reply week2/Sort/SortingAndSearching.java]: fixed

**Review by VBhaskarachary — APPROVED** [review]
Changes looks good to me

**Comment by Kukkadapu-Ruchitha** [comment]
abstraction,interface,pairedsum,sorting

**Pull request merged**